src/PostgREST/ApiRequest.hs

```diff
@@ -124,13 +125,13 @@ userApiRequest schema req reqBody
   payload =
     case decodeContentType . fromMaybe "application/json" $ lookupHeader "content-type" of
       CTApplicationJSON ->
-          either Left (\val -> case ensureUniform (pluralize val) of
-            Nothing -> Left "All object keys must match"
-            Just json -> Right json) (JSON.eitherDecode reqBody)
+        (if BL.null reqBody && isTargetingProc
```

Comment:
Sure, this spares us also the parentheses. I find it a bit more readable though to have the reverse bind on the second line e.g.:
```
note "All lines must have same number of fields" . ensureUniform . csvToJson
          =<< CSV.decodeByName reqBody
```
This makes it immediately clear which operator combines the two lines.